datastore/adapter/keytransform.py

```diff
@@ -290,3 +295,161 @@ class BinaryNestedPathAdapter(_NestedPathAdapter[datastore.abc.BinaryDatastore],
 
 class ObjectNestedPathAdapter(_NestedPathAdapter[datastore.abc.ObjectDatastore], datastore.abc.ObjectAdapter):  # noqa: E501
 	...
+
+
+class _FlatFSAdapter(_Adapter[DS], typing.Generic[DS]):
+	"""Represents a simple DatastoreAdapter that shards/namespaces incoming keys.
+
+	Incoming keys are sharded into namespaces according to specifications
+	in a `SHARDING` file, which `prefix`, `suffix`, and `next-to-last`
+	sharding functions available.
+
+	The file should be be a single line in the following format:
+		{prefix}/version/function_name/key_length
+	
+	Implementation Notes:
+		* The `default_prefix` is "/repo/flatfs/shard/"
+		* The only version supported is "v1"
+		* function_name must be `prefix`, `suffix`, or `next-to-last`
+
+	For example:
+
+		>>> import datastore
+		>>>
+		>>> ds = datastore.DictDatastore()
+		>>> np = datastore.FlatFSDatastore(ds)
+		>>>
+		>>> np.put(datastore.Key('/abcdefghijk'), 1)
+		>>> np.get(datastore.Key('/abcdefghijk'))
+		1
+		>>> ds.get(datastore.Key('/abcdefghijk'))
+		None
```

Comment:
Note: Actually it would raise KeyError here

Reply:
Blindly copied from the outdated ones in `NestedPath`. Would appreciate guidance regarding my [comment](https://github.com/ipfs/py-datastore/issues/16#issuecomment-581173361) on #16 and then will fix up.

Reply:
You're right, they're all wrong either way! :stuck_out_tongue: 

Unfortunately, for now `trio.run(lambda: …)` will be the best we can do however.